eclipse-collections-code-generator/src/main/resources/api/block/predicate/primitivePredicate.stg

```diff
@@ -23,8 +23,15 @@ import java.io.Serializable;
  * @since 3.0.
  */
 public interface <name>Predicate
-        extends Serializable
+        extends <if(primitive.specializedStream)>java.util.function.<name>Predicate, <endif>Serializable
 {
+<if(primitive.specializedStream)>
+    default boolean test(<type> value)
+    {
+        return this.accept(value);
+    }
+
+<endif>
```

Comment:
Good point. For consistency, I should also have the @FunctionalInterface and @Override annotations.